05/script.js

```diff
@@ -0,0 +1,45 @@
+function Student(initFirstName, initLastName) {
+	this.name = initFirstName
+	this.lastName = initLastName
+	this.marks = []
+}
+
+// dodanie oceny
+Student.prototype.getMark = function (subject, mark) {
+	if (typeof this.marks[subject] === 'undefined') {
+		this.marks[subject] = []
+	}
+	this.marks[subject].push(mark)
+}
+
+// średnia z przedmiotu
+Student.prototype.getAverageGrade = function (subject) {
+	if (typeof subject === 'undefined') {
+	}
+	const grades = this.grades[subject]
+}
+
+// srednia cała
+
+Student.prototype.avgAll = function () {
+	const gradesSums = this.marks[subject]
+	let sumMarks = 0
+	let sumSubject = 0
+
+	for (const grade of gradesSums) {
+		const sumMarskAll = this.sumMarskAllThis(grade)
+		sumMarks += sumMarskAll
+		sumSubject++
+	}
+	return sumSubject === 0 ? 0 : sumMarks / sumSubject
```

Comment:
Tutaj musimy obliczyć sumę wszystkich ocen, aby to zrobić to potrzebujemy mieć dostęp do nich wszystkich.
Obecnie trochę mylnie nazwałeś sobie zmienne w pętli bo `gradesSums` sugeruje, że mamy  oceny, natomiast to co jest przypisane do tej zmiennej to `this.marks[subject]`, gdzie `subject` nie występuje i już w linii 25 jest błąd.

Dodatkowo `for of` operuje raczej na elementach tablicy, a nie objektu dlatego też może miałeś z tym problem.

Tutaj powinniśmy dostać się do ocen dla danego przedmiotu i dopiero potem obliczać sumę i średnią - można to osiągnąć w taki sposób:
```
Student.prototype.avgAll = function () {
	console.log(this.marks);
	for(const subject in this.marks) { // objekt dlatego for in
		const subjectMarks = this.marks[subject];
		for(const index  in subjectMarks ) {
			const mark = subjectMarks[index];
			console.log(mark);
		}

		//lub

		for(const mark of subjectMarks) {
			console.log(mark, 'from: for of loop')
		}
	}
}
```